core/storage/page_cache.rs

```diff
@@ -276,7 +277,11 @@ impl DumbLruPageCache {
         while need_to_evict > 0 && current_opt.is_some() {
             let current = current_opt.unwrap();
             let entry = unsafe { current.as_ref() };
-            current_opt = entry.prev; // Pick prev before modifying entry
+            // Pick prev before modifying entry
+            current_opt = entry.prev;
+            if entry.page.is_pinned() {
+                continue;
+            }
```

Comment:
wait there cowboy, does this mean we missed `if is_dirty { continue }` check?

Reply:
the page is not necessarily dirty, since we just want to pin the pages we have pushed onto the stack to guarantee they are in memory if child needs to check parent status